docs/source/conf.py

```diff
@@ -12,6 +12,7 @@
 #
 import os
 import sys
+sys.path.append(os.path.abspath("./_ext"))
 sys.path.insert(0, os.path.abspath('.'))
 
```

Comment:
Why are these PATH manipulations necessary, i.e., the `sys.path ....` statements?

Reply:
Hello @ajpowelsnl `docs/source/conf.py` is a Sphinx config file. In this file one could customize Sphinx behavior. This is a Python file as well, so things defined there go through Python's interpreter.
Line `sys.path.append(os.path.abspath("./_ext"))` adds directory `_ext` to the path. This is where out extension `cppkokkos` is located (`_ext/cppkokkos.py`).
Extension is added a few lines below:
```Python3
extensions = ["myst_parser",
              "sphinx.ext.autodoc",
              "sphinx.ext.viewcode",
              "sphinx.ext.intersphinx",
              "sphinx_copybutton",
              "cppkokkos"]
```

Summarizing: this allows us to use our custom extension `cppkokkos` with Sphinx, where new directives are added and other are customized.
I hope that answer your question.